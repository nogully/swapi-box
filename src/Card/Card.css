.Card {
  background-color: #0A0847;
  border: 1px solid #0A0847;
  height: 350px;
  margin: 10px;
  opacity: .8;
  padding: 0px 10px;
  width: 250px;
}

.active {
  background-color: purple;
}

.Card h1 {
  color: #feda4a;
}