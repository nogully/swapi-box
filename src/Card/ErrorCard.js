import React from 'react';

const ErrorCard = ({ data, favoriteCard, isActive, key }) => {
  const attributes = Object.keys(data).filter( attribute => attribute !== 'name')
  const elements = attributes.map( (attribute, index) => {
    return <p key={index}>{`${attribute} : ${data[attribute]}`}</p>
  })

  return (
    <article className={ "Error " } >
      <h1>Error</h1>
      <p>There are no favorites</p>
    </article>
  )

}

export default ErrorCard;
