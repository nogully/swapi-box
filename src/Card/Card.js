import React from 'react';
import './Card.css'
import PropTypes from 'prop-types'

const Card = ({ data, favoriteCard, isActive }) => {
  const attributes = Object.keys(data).filter( attribute => attribute !== 'name')
  const elements = attributes.map( (attribute) => {
    return <p key={attribute}>{`${attribute} : ${data[attribute]}`}</p>
  })

  return (
    <article className={ "Card " + isActive } >
      <h1>{ data.name }</h1>

      { elements }
      
      <button onClick={() => favoriteCard(data)} 
              className='light'> 
        favorite
      </button>
    </article>
  )

}

Card.propTypes = {
  data: PropTypes.object.isRequired,
  favoriteCard: PropTypes.func.isRequired,
  isActive: PropTypes.string.isRequired, 
}

export default Card;
