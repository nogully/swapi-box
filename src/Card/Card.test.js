 /* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import { shallow, mount } from 'enzyme';

describe('Card', () => {
  let wrapper
  let mockObj 
  let favoriteCard

  beforeEach(() => {
    mockObj = { "name":'Endor', 
                "terrain": 'forest', 
                "climate": 'temperate', 
                "population": '30000000',
                "residents": 'Ewoks' }
    favoriteCard = jest.fn()
    wrapper = shallow(<Card data={ mockObj } 
                            favoriteCard={ favoriteCard } 
                            isActive='active'/>);
  });

  it('exists and matches snapshot', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('will give a class of active to cards in the favorites array', () => {
    expect(wrapper.find('.active').length).toEqual(1);
  })

  it('should call favoriteCard() with the button element', () => {
    wrapper.find('button').first().simulate('click');
    expect(favoriteCard).toHaveBeenCalledWith(mockObj)
  })

})