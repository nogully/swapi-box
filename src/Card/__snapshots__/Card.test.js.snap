// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card exists and matches snapshot 1`] = `
<article
  className="Card active"
>
  <h1>
    Endor
  </h1>
  <p
    key="terrain"
  >
    terrain : forest
  </p>
  <p
    key="climate"
  >
    climate : temperate
  </p>
  <p
    key="population"
  >
    population : 30000000
  </p>
  <p
    key="residents"
  >
    residents : Ewoks
  </p>
  <button
    className="light"
    onClick={[Function]}
  >
    favorite
  </button>
</article>
`;
